price-estimator/src/solver_rounding_buffer.rs

```diff
@@ -45,15 +46,15 @@ pub fn apply_rounding_buffer(
     // Apply rounding buffer to account balances and order sell amounts.
     for order in orders.iter_mut() {
         // If a token doesn't have a price it means it isn't connected to the fee. In this case we
-        // use a price of 1 to get some reasonable default rounding buffer.
+        // reuse the fee token price to get some reasonable default rounding buffer.
         let (sell_token, buy_token) = (TokenId(order.sell_token), TokenId(order.buy_token));
         let buy_token_price = match token_prices.get(&buy_token) {
             Some(price) if *price > 0 => *price as f64,
-            _ => 1.0,
+            _ => fee_token_price,
```

Comment:
Should the default consider the token's decimals? Depending on if a higher price leads to a larger or lesser rounding buffer we should probably choose a price of 10^^(decimals +/- 5 or the like). This would allow to cover prices from 0,0001 to 10000

Reply:
Assuming that tokens usually have decimals set so that one base unit is very roughly one USD this would make sense.
I'm thinking a good way to sidestep this here would be to instead of taking the HashMap with prices take a function `TokenId -> Price`, then the caller has to decide how they want to implement it.
In a future PR that has to call this function we can then do something like what you describe.

Reply:
Implemented this idea as another commit in this PR. The discussion is still valid but this module won't have to make the decision.

Reply:
We might want to have different default behaviour based on whether it's the buy or sell token. Given the buffer depends on 

`let estimated_xrate = estimated_buy_token_price / estimated_sell_token_price;`

we might want the buy_token_price fallback to something large (10^(decimals+5)) and the sell_token_price fall back to something small (10^(decimals-5))

I think to fine-tune this it would be nice to see what the impact of these changes would be on accuracy. But for now, I think it's fine assuming 10^decimals (we should document somewhere all the places on which we might fine-tune later).